pkg/telemetry/context.go

```diff
@@ -0,0 +1,53 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type rpcContextName string
+
+const (
+	methodContextName  rpcContextName = "method"
+	serviceContextName rpcContextName = "service"
+)
+
+func saveMethodServiceInContext(ctx context.Context, method string, service string) context.Context {
+	methods := strings.Split(method, "/")
+
+	ctx = context.WithValue(ctx, methodContextName, methods[len(methods)-1])
+	return context.WithValue(ctx, serviceContextName, service)
+}
+
+// ContextWithRPCInfo will save the rpc method and service information in context
+func ContextWithRPCInfo(ctx context.Context) context.Context {
+	method, found := grpc.Method(ctx)
+	if !found {
+		method = "unknown"
+	}
+	return saveMethodServiceInContext(ctx, method, openfgav1.OpenFGAService_ServiceDesc.ServiceName)
+}
+
+func getContextInfo(ctx context.Context, key rpcContextName) string {
```

Comment:
This extra function is redundant. This is the responsibility of the context getter anyways `RPCInfoFromContext`. The extra function just adds indirection for no value.

Reply:
This method is to provide sane behaviour even if context is not set properly